lib/usecases/base/base_use_case.dart

```diff
@@ -23,3 +23,9 @@ abstract class SimpleUseCase<T> {
 
   T call();
 }
+
+abstract class StorageUseCase<T> {
+  const StorageUseCase();
+
+  void save(T data);
+}
```

Comment:
IMO, the naming is too specific to be inside `base_use_case.dart`

Could we rename it to something more general? 💭 

For example:

```
abstract class SimpleInputUseCase<I> {
  const SimpleInputUseCase();

  void call(I input);
}
```

Reply:
Renamed it to `DataSaverUseCase`